addons/vscode/src/extension.ts

```diff
@@ -493,7 +495,7 @@ const launchPreview = async (task: LaunchInBrowserTask | LaunchInWebViewTask) =>
 		html = html.replace(
 			"preview-arg:previewMode:Doc",
 			`preview-arg:previewMode:${previewMode}`
-		);
+		).replace("VSCODE_STATE", JSON.stringify({ mode: task.mode, fsPath: bindDocument.uri.fsPath }));
```

Comment:
It looks like we are using a replace trick, but `VSCODE_STATE` looks easy to occur in code elsewhere. To make it  more robust, we can make some construction like:

```
.replace("preview-arg:state:{}", `preview-arg:state:{JSON.stringify({ mode: task.mode, fsPath: bindDocument.uri.fsPath })}`)
```

Please also make sanity checking on `bindDocument.uri.fsPath` to avoid use cracking us. I imagine both quote mark and backtick mark are valid characters in a path:
```js
.replace("VSCODE_STATE", JSON.stringify({ mode: task.mode, fsPath: "/I'm cracking you/` + (alert(`message`)) + `" }))
```

It is also arguable whether we can have a better solution to pass arguments to frontend.

Reply:
OK, sounds good!
Using the preview-arg:state name, how would you get the object itself?